src/refs.c

```diff
@@ -1455,3 +1455,18 @@ int git_reference__resolve_unborn(git_reference **out_ref, git_repository *repo,
 
 	return error;
 }
+
+int git_reference__is_unborn(const git_reference *ref, git_repository *repo)
+{
+	int error;
+	assert(ref && repo);
+
+	if (ref->type == GIT_REF_OID)
+		return 0;
+
+	error = git_reference__resolve_unborn(NULL, repo, ref);
+	if (error != 0 && error != GIT_EUNBORNBRANCH)
+		return error;
+
+	return error == GIT_EUNBORNBRANCH ? true : false;
```

Comment:
I think we usually avoid reusing the return code as value and error code. So maybe use an out parameter instead to return the boolean?